[Code Clean] Replace `std::runtime_error` with `TORCH_CHECK`

KarhouTam opened the pull request:
Including:
- `torch/csrc/instruction_counter`
- `torch/csrc/lazy`
- `torch/csrc/monitor`
- `torch/csrc/profiler`
- `torch/csrc/dynamo`

Fixes part of #148114 

Personal mistake about (PR #163317), this PR does the same thing **and PR #163317 has already been approved by @albanD.**

This is a personal mistake on my part, and I'm so sorry about that. Hope you won't mind @albanD. 🥹

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

albanD reviewed (approved):
No worries at all!

Skylion007 reviewed (approved):
Some last minute nits, but looks good once those are fixed.

KarhouTam commented on torch/csrc/monitor/counters.cpp:
```diff
@@ -21,7 +22,8 @@ const char* aggregationName(Aggregation agg) {
-      throw std::runtime_error(
+      TORCH_CHECK(
+          false,
           "unknown aggregation: " + std::to_string(static_cast<int>(agg)));
```
Got it. Thanks

cyyever commented:
@pytorchbot merge

KarhouTam commented:
@pytorchbot merge

KarhouTam commented:
@pytorchbot merge

cyyever commented:
@pytorchbot merge

KarhouTam committed:
[Code Clean] Replace `std::runtime_error` with `TORCH_CHECK`

Including:
- `torch/csrc/instruction_counter`
- `torch/csrc/lazy`
- `torch/csrc/monitor`
- `torch/csrc/profiler`
- `torch/csrc/dynamo`

KarhouTam commented:
@pytorchbot merge
